### DISCUSSION TITLE
TuyaMCU blinds: Prevent "Dimmer 0" from setting to OFF?

### DISCUSSION THREAD

**Discussion by microlomaniac** [arendst/Tasmota#11901] [Q&A] [answered]
Hi,

I have successfully tasmotized a [Zemismart M515EGB blinds motor](https://templates.blakadder.com/zemismart_M515EGB.html) with Tasmota 9.4.0, and I've integrated it successfully into my Homebridge setup with [Homebridge Mqqtthing](https://github.com/arachnetech/homebridge-mqttthing/).

The TuyaMCU dpid is mapped to Dimmer via `TuyaMCU 21,2`. All values except 0 work fine. But when I try to close the blinds (set dimmer value to 0), the "Toggle" value gets set to OFF and the motor does not budge.

After setting `SetOption77 1`, it works as expected in the WebUI, but controlling it via Homebridge still does not.

My Homebridge Config for this accessory (excerpt):
```
            "accessory": "mqttthing",
            "type": "windowCovering",
            "topics": {
                "getCurrentPosition": {
                    "topic": "stat/tasmota_XXXXF0/RESULT",
                    "apply": "return JSON.parse(message).Dimmer;"
                },
                "setTargetPosition": "cmnd/tasmota_XXXXF0/Dimmer",
                "getTargetPosition": {
                    "topic": "stat/tasmota_XXXXF0/RESULT",
                    "apply": "return JSON.parse(message).Dimmer;"
                },
                "getPositionState": {
                    "topic": "stat/tasmota_XXXXF0/RESULT",
                    "apply": "return JSON.parse(message).Dimmer;"
                }
            },
            "positionStateValues": [
                "-1",
                "1",
                "0"
            ],
            "startPub": {
                "cmnd/tasmota_XXXXF0/shutterposition1": ""
            }
```

Is this a bug, or am I missing something?

**Comment by barbudor** [comment]
I'm not very familiar with blinds and shutters but it seems odd to me to use a Dimmer feature to control a motor position
I understand you did what is listed in the template page

But Dimmer 0 is actually a command to switch off a light, so it is normal and expected that the toggle turns off and no motion.
You probably need to limit to the 1..100 range

**Reply by microlomaniac** [reply]
Using the Dimmer feature seems to be the only available option when interfacing with a TuyaMCU, as per the [Tasmota documentation](https://tasmota.github.io/docs/TuyaMCU/).
I have not found a way to combine the [Shutters and Blinds](https://tasmota.github.io/docs/Blinds-and-Shutters/) feature with TuyaMCU.

Unfortunately, limiting the DimmerRange to 1..100 didn't help, as the MQTT command still requests "0":
`CMD: Group 0, Index 1, Command "DIMMER", Data "0"`
which is "overinterpreted" to also set POWER to OFF.

I guess it would help limiting the range within the Homebridge MQTTthing plug-in, but I am not aware of a way to do that.

**Reply by sfromis** [reply]
You might be able to get around the issue by replacing the Dimmer command with a command like
`event setTargetPosition=` (value)
and have rules like
`on event#setTargetPosition==0 do Dimmer 1 endon`
`on event#setTargetPosition>0 do Dimmer %value% endon`
Not knowing anything about that Homebridge config scheme, I can't suggest howto there.

**Reply by microlomaniac** [reply]
Thanks for the tip, sfromis.
I have now enabled this rule:
`on Power1#state=0 do Dimmer 1 endon`
and it is a workaround, but far from ideal.

While the blinds now close to 1%, which is close enough, it is still displayed on my iDevices as "1% open", (accurate, obviously) so when I press the button, it sends the "close" command, which never gets the device to the "closed" state. (thus there is no button to simply press to open the blinds. Quickly double-tapping works, though)

I have tried
`on Dimmer#state=0 do Power1 1 endon`
hoping that it would then move to the target position, but that didn't work.

**Reply by sfromis** [reply]
If you want more control over what you'll see in Homebridge, you can in a similar way replace the use of the Dimmer command with event commands where you can have rules return the values you like to have. If the structure with the event command and a keyword in the payload is not easy to setup, you can also make "new" Tasmota commands using the [subscribe](https://tasmota.github.io/docs/Subscribe-%26-Unsubscribe/) feature.

**Comment by microlomaniac** [comment]
Additional info:

Entering `Dimmer 0` in the console leads to the same unwanted behaviour.
Result is `{"POWER":"OFF","Dimmer":0}` which leads to the motor not moving.

**Comment by microlomaniac** [comment, marked as answer]
So I dug into the code, #defined `DEBUG_LIGHT` and found `TYA: Send dim skipped value 0 for dpid 2`.

Thusly, I looked it up, found the culprit in [xdrv_16_tuyamcu.ino](https://github.com/arendst/Tasmota/blob/development/tasmota/xdrv_16_tuyamcu.ino), line 639, and removed the `if (duty > 0)` part as follows:

```
    if (/*duty > 0 && */ !Tuya.ignore_dim && TuyaSerial && dpid > 0) {      
``` 

It seems odd to me that sending a dimmer command with a duty of 0 is strictly prohibited, but it must have its reasons.

Anyway, all is well now! No rules, `Dimmer 0` works, Homebridge integration is fine.
